frame/benchmarking/src/lib.rs

```diff
@@ -764,120 +812,68 @@ macro_rules! impl_benchmark {
 					// Reset the read/write counter so we don't count operations in the setup process.
 					$crate::benchmarking::reset_read_write_count();
 
-					if verify {
-						closure_to_benchmark()?;
-					} else {
-						// Time the extrinsic logic.
-						$crate::log::trace!(
-							target: "benchmark",
-							"Start Benchmark: {:?}", c
-						);
+					// Time the extrinsic logic.
+					$crate::log::trace!(
+						target: "benchmark",
+						"Start Benchmark: {:?}", c
+					);
 
-						let start_pov = $crate::benchmarking::proof_size();
-						let start_extrinsic = $crate::benchmarking::current_time();
+					let start_pov = $crate::benchmarking::proof_size();
+					let start_extrinsic = $crate::benchmarking::current_time();
 
-						closure_to_benchmark()?;
+					closure_to_benchmark()?;
 
-						let finish_extrinsic = $crate::benchmarking::current_time();
-						let end_pov = $crate::benchmarking::proof_size();
+					let finish_extrinsic = $crate::benchmarking::current_time();
+					let end_pov = $crate::benchmarking::proof_size();
 
-						// Calculate the diff caused by the benchmark.
-						let elapsed_extrinsic = finish_extrinsic.saturating_sub(start_extrinsic);
-						let diff_pov = match (start_pov, end_pov) {
-							(Some(start), Some(end)) => end.saturating_sub(start),
-							_ => Default::default(),
-						};
+					// Calculate the diff caused by the benchmark.
+					let elapsed_extrinsic = finish_extrinsic.saturating_sub(start_extrinsic);
+					let diff_pov = match (start_pov, end_pov) {
+						(Some(start), Some(end)) => end.saturating_sub(start),
+						_ => Default::default(),
```

Comment:
when can this fail? isn't it better to stop rather than mistakenly provide a zero diff for pov?

Reply:
will patch in new pr

Reply:
https://github.com/paritytech/substrate/issues/9513

Reply:
actually some pallets dont give pov data yet